src/basic/pidref.c

```diff
@@ -58,15 +58,27 @@ bool pidref_equal(PidRef *a, PidRef *b) {
                 if (a->fd_id == 0 || b->fd_id == 0)
                         return true;
         } else {
+                /* If this is the very same structure and this is local, it definitely refers to the same process */
+                if (a == b)
```

Comment:
Hmm, even if the pidref is remote the same structure would very much mean the same process? Why conditionalize on local pidrefs?